frontend/pages/hosts/ManageHostsPage/components/HostsDataTable/HostsDataTable.jsx

```diff
@@ -1,78 +1,217 @@
+      { Header: 'Status', disableSortBy: true, accessor: 'status', Cell: cellProps => <StatusCell value={cellProps.cell.value} /> },
+      { Header: cellProps => <HeaderCell all={cellProps.column} value={'OS'} isSortedDesc={cellProps.column.isSortedDesc} />, accessor: 'os_version', Cell: cellProps => <TextCell value={cellProps.cell.value} /> },
+      { Header: cellProps => <HeaderCell value={'Osquery'} isSortedDesc={cellProps.column.isSortedDesc} />, accessor: 'osquery_version', Cell: cellProps => <TextCell value={cellProps.cell.value} /> },
+      { Header: cellProps => <HeaderCell value={'IPv4'} isSortedDesc={cellProps.column.isSortedDesc} />, accessor: 'primary_ip', Cell: cellProps => <TextCell value={cellProps.cell.value} /> },
+      { Header: cellProps => <HeaderCell value={'Physical Address'} isSortedDesc={cellProps.column.isSortedDesc} />, accessor: 'primary_mac', Cell: cellProps => <TextCell value={cellProps.cell.value} /> },
+      { Header: 'CPU', disableSortBy: true, accessor: 'host_cpu', Cell: cellProps => <TextCell value={cellProps.cell.value} /> },
+      { Header: cellProps => <HeaderCell value={'Memory'} isSortedDesc={cellProps.column.isSortedDesc} />, accessor: 'memory', Cell: cellProps => <TextCell value={cellProps.cell.value} formatter={humanHostMemory} /> },
+      { Header: cellProps => <HeaderCell value={'Uptime'} isSortedDesc={cellProps.column.isSortedDesc} />, accessor: 'uptime', Cell: cellProps => <TextCell value={cellProps.cell.value} formatter={humanHostUptime} /> },
     ];
   }, []);
 
-  const data = React.useMemo(() => {
-    return hosts;
-  }, [hosts]);
+  const data = useMemo(() => {
+    return hostAPIOrder.map((id) => {
+      return hosts[id];
+    });
+  }, [hosts, hostAPIOrder]);
 
   const {
     headerGroups,
     rows,
     prepareRow,
-  } = useTable({ columns, data });
+    setGlobalFilter,
+    state: tableState,
+  } = useTable(
+    { columns,
+      data,
+      initialState: {
+        sortBy: [{ id: DEFAULT_SORT_KEY, desc: true }],
+      },
+      disableMultiSort: true,
+      manualGlobalFilter: true,
+      manualSortBy: true,
+      autoResetSortBy: !skipPageResetRef.current,
+      autoResetGlobalFilter: !skipPageResetRef.current,
+    },
+    useGlobalFilter,
+    useSortBy,
+  );
+  const { globalFilter, sortBy } = tableState;
+
+  const debouncedGlobalFilter = useAsyncDebounce((value) => {
+    skipPageResetRef.current = true;
+    setGlobalFilter(value || undefined);
+  }, DEBOUNCE_QUERY_DELAY);
+
+  const onPaginationChange = useCallback((newPage) => {
+    if (newPage > pageIndex) {
+      // pageIndexChangeRef.current = pageIndex;
+      setPageIndex(pageIndex + 1);
+    } else {
+      // pageIndpageIndexChangeRefexRef.current = pageIndex;
+      setPageIndex(pageIndex - 1);
+    }
+    pageIndexChangeRef.current = true;
+    // scrollToTop();
```

Comment:
Intended to leave this code commented here?

Reply:
got to add that back

Reply:
added